src/mappings/handlers/evmHandlers.ts

```diff
@@ -9,13 +11,314 @@ import { InvestorTransactionData, InvestorTransactionService } from '../services
+    reserve: [{ address: '0xc156aa23a29ef7678b4dec5b812b84797ed21ee0', startBlock: 15460175 }],
+    shelf: [{ address: '0x511680e8eD9cA463E5F1776281992de4Aec22039' }],
+    pile: [{ address: '0x8f7b701a7309c2Ef5c0Fe850ab7589d5b23a8f76' }],
+  },
+  {
+    id: '0xdb3bc9fb1893222d266762e9ff857eb74d75c7d6',
+    shortName: 'ConsolFreight 4',
+    startBlock: 11063000,
+    navFeed: [{ address: '0x69504da6b2cd8320b9a62f3aed410a298d3e7ac6', startBlock: 11063046 }],
+    reserve: [
+      { address: '0x0d601b451aFD502e473bA4CE6E3876D652BCbee7', startBlock: 11063150 },
+      { address: '0xfaec38ffee969cf18e88097ec62e30b70494e234', startBlock: 13004061 },
+    ],
+    shelf: [{ address: '0xA0B0d8394ADC79f5d1563a892abFc6186E519644' }],
+    pile: [{ address: '0x3fC72dA5545E2AB6202D81fbEb1C8273Be95068C' }],
+  },
+  {
+    id: '0xf96f18f2c70b57ec864cc0c8b828450b82ff63e3',
+    shortName: 'ALT 1.0',
+    startBlock: 14167745,
+    navFeed: [{ address: '0x6fb02533b264d103b84d8f13d11a4865ec96307a', startBlock: 14167760 }],
+    reserve: [{ address: '0x35f7de72c8597599ef4f2a18eafd162257301435', startBlock: 14167898 }],
+    shelf: [{ address: '0x11daC3fA9d2216377A79Bef04F6A2682630371c3' }],
+    pile: [{ address: '0xE18AAB16cC26EB23740D72875e0C6b52cEbb46b3' }],
+  },
+  {
+    id: '0xfc2950dd337ca8496c18dfc0256fb905a7e7e5c6',
+    shortName: 'databased.FINANCE 1',
+    startBlock: 11639815,
+    navFeed: [{ address: '0x00cd3ae59fdbd375a187bf8074db59edaf766c19', startBlock: 11639830 }],
+    reserve: [{ address: '0x729e12cdc0190a2e4ab4401bca4c16132d75adc5', startBlock: 11639877 }],
+    shelf: [{ address: '0xC42CfB07bC1140f9A615bD63c4fFAE5F8260Ab22' }],
+    pile: [{ address: '0xdB07B21109117208a0317adfbed484C87c9c2aFf' }],
+  },
+]
+
+export const handleEvmBlock = errorHandler(_handleEvmBlock)
+async function _handleEvmBlock(block: EthereumBlock): Promise<void> {
+  const date = new Date(Number(block.timestamp) * 1000)
+  const blockNumber = block.number
+  const newPeriod = (await timekeeper).processBlock(date)
+
+  const DAIAddress = '0x6b175474e89094c44da98b954eedeac495271d0f'
+  const blockchain = await BlockchainService.getOrInit('1')
+  const currency = await CurrencyService.getOrInitEvm(blockchain.id, DAIAddress)
+
+  if (newPeriod) {
+    logger.info(`It's a new period on EVM block ${blockNumber}: ${date.toISOString()}`)
+
+    // update pool states
+    for (const tinlakePool of pools) {
+      let pool
+      if (block.number >= tinlakePool.startBlock) {
+        pool = await PoolService.getOrSeed(tinlakePool.id)
+        if (block.number >= tinlakePool.startBlock && pool.totalReserve == null) {
+          pool.totalReserve = BigInt(0)
+          pool.portfolioValuation = BigInt(0)
+          pool.currency
+          pool.isActive = false
+          pool.currencyId = currency.id
+          await pool.save()
+          logger.info(`Initializing pool ${tinlakePool.id}`)
+        }
+        const latestNavFeed = tinlakePool.navFeed.reduce(
```

Comment:
Would be nice to add a helper for this reduce, something like `getLatestContract(tinlakePool.navFeed, current.startBlock)`